frontend/lib/src/dataframes/arrowParseUtils.ts

```diff
@@ -195,182 +121,286 @@ function parseIndexData(table: Table, pandasSchema: PandasSchema): IndexData {
+ * Parse Arrow bytes (IPC format) into a couple of components
+ * that allow easier and more efficient access to the data.
  *
  * @param ipcBytes - Arrow bytes (IPC format)
- * @returns - Parsed Arrow table split into different
- *  components for easier access: columnNames, fields, indexData, indexNames, data, columnTypes.
+ * @returns - Parsed table components.
  */
 export function parseArrowIpcBytes(
   ipcBytes: Uint8Array | null | undefined
 ): ParsedTable {
-  // Load arrow table object from IPC data
+  // Load arrow table object from Arrow IPC bytes.
+  // The table contains all the cell data, the arrow schema
+  // and the pandas schema (if processed through Pandas).
   const table = tableFromIPC(ipcBytes)
-  // Load field information for all columns:
-  const fields = parseFields(table.schema)
 
-  // Load pandas schema from metadata (if it exists):
-  const pandasSchema = parsePandasSchema(table)
+  // The arrow schema contains type information for all columns
+  // that are part of the table. This doesn't include range indices
+  // which are only part of the pandas schema and need to be parsed
+  // separately below.
+  const arrowSchema = table.schema
 
-  // Load all column names from table schema:
-  const columnNames = parseColumnNames(pandasSchema)
+  // Load pandas schema from metadata.
+  // Pandas schema only exists if the table was processed through Pandas.
+  const pandasSchema = parsePandasSchema(table)
 
-  // Load the display names of the index columns:
-  const indexNames = parseIndexNames(pandasSchema)
+  // Load categorical options for each column that has a categorical type.
+  // This is a mapping of column names to categorical options that is
+  // used in a later step to attach to the column type information.
+  const categoricalOptions = parseCategoricalOptionsForColumns(table)
+
+  // Load the type information for index columns.
+  // Index columns refer to the row index (row labels) of a Pandas DataFrame:
+  // https://pandas.pydata.org/docs/user_guide/indexing.html
+  // Therefore, index columns are only present if the
+  // table was processed through Pandas.
+  const indexColumnTypes = parseIndexColumnTypes(
+    arrowSchema,
+    pandasSchema,
+    categoricalOptions
+  )
 
-  // Extract unprocessed column names from pandas schema
-  // (needed for parsing the data cells below):
-  const rawColumns = getRawColumns(pandasSchema)
+  // Load the type information for data columns:
+  const dataColumnTypes = parseDataColumnTypes(
+    arrowSchema,
+    pandasSchema,
+    categoricalOptions
+  )
 
-  // Load all non-index data cells:
-  const data = parseData(table, columnNames, rawColumns)
+  // Load all cell data for data columns:
+  const data = parseData(table, dataColumnTypes)
 
-  // Load all index data cells:
+  // Load all cell data for index columns.
+  // Will be empty if the table was not processed through Pandas.
   const indexData = parseIndexData(table, pandasSchema)
```

Comment:
nit: the comment makes it sound like this is Pandas only. If so, the variable and function names should be renamed similar to the other renamings you did, e.g. `pandasIndexData` and `parsePandasIndexData`

Reply:
Renamed it 👍